anchor-service/src/anchor_batch.rs

```diff
@@ -0,0 +1,118 @@
+use std::{sync::Arc, time::Duration};
+
+use anyhow::Result;
+use async_trait::async_trait;
+use cid::Cid;
+use indexmap::IndexMap;
+use tokio::time::interval;
+use tracing::{error, info};
+
+use crate::{
+    anchor::{AnchorRequest, TimeEventBatch},
+    merkle_tree::{build_merkle_tree, MerkleTree},
+    time_event::build_time_events,
+    transaction_manager::{Receipt, TransactionManager},
+};
+
+/// An AnchorClient is responsible for fetching AnchorRequests and storing TimeEvents.
+#[async_trait]
+pub trait Store: Send + Sync {
+    /// Get a batch of AnchorRequests.
+    async fn get_anchor_requests(&self) -> Result<Vec<AnchorRequest>>;
+    /// Store a batch of TimeEvents.
+    async fn put_time_events(&self, batch: TimeEventBatch) -> Result<()>;
```

Comment:
The `impl Store for CeramicEventService` is in part 5. That is where I would like to make the decision on where we hydrate to time events and who is responsible for storing the high water mark. As part 5 is written today the high water mark is stored in the event service we will probably store that somewhere else.

`get_anchor_request` is renamed  `local_sourced_data_events` in part 3 but the real logical isolation will probably be in part 5 when the decision about where to store the high water mark drives the final trait definition.

https://github.com/ceramicnetwork/rust-ceramic/pull/488/files#diff-148a932282cead8137f562abc4a57762ed173ccebd8d74361766ec0df7948d22R424

`$1`  new_high_water_mark (i64)
```sql
INSERT INTO ceramic_one_anchor_high_water_mark (high_water_mark) VALUES ($1);
```

`$1` did:key for the local node
`$2` the limit
```sql
SELECT order_key, init_cid, cid, rowid
FROM ceramic_one_event
WHERE
  source = $1
  AND ROWID > (SELECT MAX(high_water_mark) FROM ceramic_one_anchor_high_water_mark)
ORDER BY rowid
LIMIT $2;
```

Reply:
>The impl Store for CeramicEventService is in part 5. That is where I would like to make the decision on where we hydrate to time events and who is responsible for storing the high water mark.

Why does the ceramic-service what to store the high water mark for the anchor service? I would expect the anchor service to have its own place to store the highwater mark separate from the ceramic-service. Perhaps the name `anchor_service::Store` is incorrect here in that it implies its the interface for storing state on behalf of the anchor service. I see this trait as a way access to the event service. Perhaps we should call this trait `EventService`? 

And do we need another mechnism for storing data on behalf of the anchor service. For example a new sqlite connection and its own anchor store struct that can store and retrieve the high water mark?

Reply:
To expand a bit on the architecture I am after...

I would like to have a set of services and each service manages its own state (i.e. database). The services can depend on each other via public APIs etc. In this PR we are adding a new service that depends on the event service. This means the anchor service needs it a way to manage its own state. It should not delegate management of its state to the event service. The anchor service has very little state that it needs to store (i.e. only the high water mark). However that still means that there should be some kind of struct private to the anchor service crate that manages storing and retrieving the high water mark. I expect this is a sqlite table, and it can be the same sqlite connection that the event service uses but the logic to store and retrieve that state lives inside the anchor service.

When we inevitably want to migrate how we store the high water mark, we will have to modify any code that access that storage directly. Its best if all of that code lives behind the anchor service API boundary. 

As the PR is currently structured its intended that logic lives inside the ceramic-service crate and I explicitly do not want that architecture.

Reply:
* Part 2: build trees from anchor requests and build Time Events from proofs.
* Part 3: use the remote CAS to make proofs from root CIDs. All the auth and api stuff to talk to unmodified CAS.
* Part 4: all the changes to EventService so it can find the events that anchor service wants to anchor.  e.g. source and StreamID for events.
* Part 5: starting the loop, command line args, persisting Time events and High Water Mark

I think we can answer those questions in part 5 this is mostly here as a `todo!(get_anchor_requests, put_time_events)`
and to have something to for the MockStore so I could implement and test trees building and Time event building.